src/jit/lsra.cpp

```diff
@@ -4301,7 +4301,7 @@ void LinearScan::updateRegStateForArg(LclVarDsc* argDsc)
             isFloat = true;
         }
 
-        if (isFloat)
+        if (isFloat && !compiler->opts.compUseSoftFP)
         {
```

Comment:
Can't you hoist this into the isFloat definition, e.g.:
```
        bool isFloat = isFloatRegType(argDsc->lvType)
                        && !compiler->opts.compUseSoftFP
#ifndef _TARGET_AMD64_
                        && !compiler->info.compIsVarArgs
#endif
                        ;
```

I also presume that with compUseSoftFP, lvIsHfaRegArg() should always be false.